internal/controllers/synthetic_devices_controller.go

```diff
@@ -235,25 +238,39 @@ func (vc *SyntheticDevicesController) MintSyntheticDevice(c *fiber.Ctx) error {
 		return fiber.NewError(fiber.StatusBadRequest, "Invalid signature provided")
 	}
 
-	err = vc.sendVirtualDeviceMintPayload(c.Context(), hash, req.VehicleNode, integration.TokenId, ownerSignature)
+	childKeyNumber := generateRandomNumber()
+
+	syntheticDeviceAddr, err := vc.sendVirtualDeviceMintPayload(c.Context(), hash, req.VehicleNode, integration.TokenId, ownerSignature, childKeyNumber)
 	if err != nil {
-		vc.log.Err(err).Msg("virtual device minting request failed")
-		return fiber.NewError(fiber.StatusInternalServerError, "virtual device minting request failed")
+		vc.log.Err(err).Msg("synthetic device minting request failed")
+		return fiber.NewError(fiber.StatusInternalServerError, "synthetic device minting request failed")
 	}
 
-	return c.Send([]byte("virtual device mint request successful"))
-}
+	vnID := types.NewDecimal(decimal.New(vid, 0))
+	syntheticDevice := &models.SyntheticDevice{
+		VehicleTokenID:    vnID,
+		IntegrationID:     integration.Id,
+		WalletChildNumber: childKeyNumber,
+		WalletAddress:     syntheticDeviceAddr,
+		MintRequestID:     vehicleNFT.MintRequestID,
```

Comment:
We're creating a _new_ meta-transaction for the virtual device, so we need to construct that row first.